src/main/scala/wdl4s/types/WdlArrayType.scala

```diff
@@ -40,13 +41,23 @@ sealed trait WdlArrayType extends WdlType {
   }
 
   override def isCoerceableFrom(otherType: WdlType): Boolean = otherType match {
-    case WdlArrayType(otherMemberType) => otherMemberType == WdlAnyType || otherMemberType.isCoerceableFrom(memberType)
+    case WdlEmptyArrayType => true
+    case WdlArrayType(otherMemberType) => memberType.isCoerceableFrom(otherMemberType)
     case _ => false
   }
 
   def asNonEmptyArrayType = WdlNonEmptyArrayType(memberType)
 }
 
+case object WdlEmptyArrayType extends WdlArrayType {
+  override val toWdlString: String = s"Array[_]"
+  override val guaranteedNonEmpty = false
+  val CanonicalValue = WdlArray(WdlEmptyArrayType, Seq.empty)
+  // No coercion is required since we'll only ever create these directly via the empty array literal [ ]
+  override protected def coercion = PartialFunction.empty
+  override val memberType = WdlVoidType
```

Comment:
`Void` is sort of like Scala's `Unit`, a type inhabited by a single value.  I think the more appropriate concept here is a "bottom type" which in Scala terms would suggest `WdlNothingType`.  Not that we necessarily have to think in Scala terms.  🙂   

```
scala> Array()
res0: Array[Nothing] = Array()
```